framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -149,304 +162,209 @@ public boolean hasDependentAnnotations() {
      * Viewpoint-adapts the dependent type annotations in the constructorType based on the
      * newClassTree.
      *
+     * <p>{@code constructorType} has been viewpoint-adapted to the call site, except that any
+     * dependent type annotations. The dependent type annotations will be viewpoint-adapted by this
+     * method.
+     *
      * @param newClassTree invocation of the constructor
-     * @param constructorType type of the constructor; is side-effected by this method
+     * @param constructorType type of the constructor invocation; is side-effected by this method
      */
-    public void viewpointAdaptConstructor(
+    public void atConstructorInvocation(
             NewClassTree newClassTree, AnnotatedExecutableType constructorType) {
         if (!hasDependentAnnotations()) {
             return;
         }
-        List<? extends ExpressionTree> args = newClassTree.getArguments();
-        viewpointAdaptExecutable(newClassTree, constructorType, args);
+        atInvocation(newClassTree, constructorType);
     }
 
     /**
      * Viewpoint-adapts a method or constructor invocation.
      *
+     * <p>{@code methodType} has been viewpoint-adapted to the call site, except that any dependent
+     * type annotations. The dependent type annotations will be viewpoint-adapted by this method.
+     *
      * @param tree invocation of the method or constructor
-     * @param methodType type of the method or constructor; is side-effected by this method
-     * @param argTrees the arguments to the method or constructor; subexpressions of {@code tree}
+     * @param methodType type of the method or constructor invocation; is side-effected by this
+     *     method
      */
-    private void viewpointAdaptExecutable(
-            ExpressionTree tree,
-            AnnotatedExecutableType methodType,
-            List<? extends ExpressionTree> argTrees) {
+    private void atInvocation(ExpressionTree tree, AnnotatedExecutableType methodType) {
         assert hasDependentAnnotations();
         Element methodElt = TreeUtils.elementFromUse(tree);
-        // The annotations on `viewpointAdaptedType` will be copied to `methodType`.
-        AnnotatedExecutableType viewpointAdaptedType =
+        // The annotations on `declaredMethodType` will be copied to `methodType`.
+        AnnotatedExecutableType declaredMethodType =
                 (AnnotatedExecutableType) factory.getAnnotatedType(methodElt);
-        if (!hasDependentType(viewpointAdaptedType)) {
+        if (!hasDependentType(declaredMethodType)) {
             return;
         }
 
-        JavaExpression receiver = JavaExpression.getReceiver(tree);
-        List<JavaExpression> argsJe = argumentTreesToJavaExpressions(tree, methodType, argTrees);
-
-        JavaExpressionContext context =
-                new JavaExpressionContext(receiver, argsJe, factory.getChecker());
-
         // methodType cannot be viewpoint adapted directly because it is the type post type variable
```

Comment:
I'm not sure what "direct" viewpoint-adaptation is; please clarify.

I am also confused about whether this comment is explaining the overall design (the "expect for dependent types" comment in the Javadoc above), the code below, or both.  Relatedly, I'm not sure of the relationship between the first and second sentences here.  Please clarify.